app/models/API1812/latePaymentPenalty/LatePaymentPenalty.scala

```diff
@@ -35,30 +34,97 @@ case class LatePaymentPenalty(principalChargeReference: String,
                               penaltyChargeReference: Option[String],
                               timeToPay: Option[Seq[TimeToPay]]) {
 
-  def hasTimeToPay(implicit appConfig: AppConfig): Boolean = {
-    timeToPay.fold(false)(_.exists(ttp => ttp.TTPStartDate.exists(!_.isAfter(DateService.now))
-      && ttp.TTPEndDate.exists(!_.isBefore(DateService.now))))
-  }
+  def hasTimeToPay(implicit appConfig: AppConfig): Boolean =
```

Comment:
Most of this is just formatting and auto-organising imports to fix the warnings.
Only HipLatePaymentPenalty is new